src/WebJobs.Script/Host/FunctionMetadataManager.cs

```diff
@@ -247,17 +242,20 @@ internal static string DeterminePrimaryScriptFile(JObject functionConfig, string
                 }
             }
 
-            if (string.IsNullOrEmpty(functionPrimary))
+            if (!string.IsNullOrEmpty(functionPrimary))
             {
-                throw new FunctionConfigurationException("Unable to determine the primary function script. Try renaming your entry point script to 'run' (or 'index' in the case of Node), " +
-                    "or alternatively you can specify the name of the entry point script explicitly by adding a 'scriptFile' property to your function metadata.");
+                return Path.GetFullPath(functionPrimary);
             }
-
-            return Path.GetFullPath(functionPrimary);
+            return null;
         }
 
         internal static string ParseLanguage(string scriptFilePath, IEnumerable<WorkerConfig> workerConfigs)
         {
+            if (string.IsNullOrEmpty(scriptFilePath))
```

Comment:
A comment on when this can happen (custom worker) might help future readers.

Reply:
done!